sklearn/_build_utils/openmp_helpers.py

```diff
@@ -0,0 +1,137 @@
+        # export CXXFLAGS="$CXXFLAGS -I/usr/local/opt/libomp/include"
+        # export LDFLAGS="$LDFLAGS -L/usr/local/opt/libomp/lib -lomp"
+        # export DYLD_LIBRARY_PATH=/usr/local/opt/libomp/lib
+        return []
+    # Default flag for GCC and clang:
+    return ['-fopenmp']
+
+
+def check_openmp_support():
+    """Check whether OpenMP test code can be compiled and run"""
+    ccompiler = new_compiler()
+    customize_compiler(ccompiler)
+
+    openmp_flags = get_openmp_flag(ccompiler)
+    if openmp_flags is None:
+        # skip the check if OpenMP support is explicitly disabled.
+        return
+
+    tmp_dir = tempfile.mkdtemp()
+    start_dir = os.path.abspath('.')
+
+    try:
+        os.chdir(tmp_dir)
+
+        # Write test program
+        with open('test_openmp.c', 'w') as f:
+            f.write(CCODE)
+
+        os.mkdir('objects')
+
+        # Compile, test program
+        ccompiler.compile(['test_openmp.c'], output_dir='objects',
+                          extra_postargs=openmp_flags)
+
+        # Link test program
+        extra_preargs = os.getenv('LDFLAGS', None)
+        if extra_preargs is not None:
+            extra_preargs = extra_preargs.split(" ")
+        else:
+            extra_preargs = []
+
+        objects = glob.glob(
+            os.path.join('objects', '*' + ccompiler.obj_extension))
+        ccompiler.link_executable(objects, 'test_openmp',
+                                  extra_preargs=extra_preargs,
+                                  extra_postargs=openmp_flags)
+
+        # Run test program
+        output = subprocess.check_output('./test_openmp')
+        output = output.decode(sys.stdout.encoding or 'utf-8').splitlines()
+
+        # Check test program output
+        if 'nthreads=' in output[0]:
+            nthreads = int(output[0].strip().split('=')[1])
+            openmp_supported = (len(output) == nthreads)
+        else:
+            openmp_supported = False
+
+    except (CompileError, LinkError, subprocess.CalledProcessError):
+        openmp_supported = False
+
+    finally:
+        os.chdir(start_dir)
+
+    err_message = (
+        "\n\n                    ***\n\n"
+        "It seems that scikit-learn cannot be built with OpenMP support.\n\n"
+        "- Make sure you have followed the installation instructions:\n\n"
+        "     "
+        "https://scikit-learn.org/dev/developers/advanced_installation.html"
+        "\n\n- If your compiler should support OpenMP but the build still "
+        "fails, please submit a bug report here:\n\n"
+        "     "
+        "https://github.com/scikit-learn/scikit-learn/issues\n\n"
+        "- If you explicitly want to build scikit-learn without OpenMP "
+        "support, you can set the following environment variable:\n\n"
+        "     SKLEARN_NO_OPENMP=TRUE\n\n"
+        "  and rerun the build command. Notice however that it will hurt the "
+        "performances of some estimators, which will run in sequential mode "
```

Comment:
I would remove "hurt the performance", people needing to build without OpenMP support should know what OpenMP does.